nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/ValidateCsv.java

```diff
@@ -0,0 +1,613 @@
+        final AtomicReference<Integer> totalCount = new AtomicReference<Integer>(0);
+        final AtomicReference<FlowFile> invalidFF = new AtomicReference<FlowFile>(null);
+        final AtomicReference<FlowFile> validFF = new AtomicReference<FlowFile>(null);
+
+        if(!isWholeFFValidation) {
+            invalidFF.set(session.create(flowFile));
+            validFF.set(session.create(flowFile));
+        }
+
+        session.read(flowFile, new InputStreamCallback() {
+            @Override
+            public void process(final InputStream in) throws IOException {
+                NifiCsvListReader listReader = null;
+                try {
+                    listReader = new NifiCsvListReader(new InputStreamReader(in), csvPref);
+
+                    // handling of header
+                    if(header) {
+                        List<String> headerList = listReader.read();
+                        if(!isWholeFFValidation) {
+                            invalidFF.set(session.append(invalidFF.get(), new OutputStreamCallback() {
+                                @Override
+                                public void process(OutputStream out) throws IOException {
+                                    out.write(print(headerList, csvPref, isFirstLine.get()));
+                                }
+                            }));
+                            validFF.set(session.append(validFF.get(), new OutputStreamCallback() {
+                                @Override
+                                public void process(OutputStream out) throws IOException {
+                                    out.write(print(headerList, csvPref, isFirstLine.get()));
+                                }
+                            }));
+                            isFirstLine.set(false);
+                        }
+                    }
+
+                    boolean stop = false;
+
+                    while (!stop) {
+                        try {
+
+                            final List<Object> list = listReader.read(cellProcs);
+                            stop = list == null;
+
+                            if(!isWholeFFValidation && !stop) {
+                                validFF.set(session.append(validFF.get(), new OutputStreamCallback() {
+                                    @Override
+                                    public void process(OutputStream out) throws IOException {
+                                        out.write(print(list, csvPref, isFirstLine.get()));
+                                    }
+                                }));
+                                okCount.set(okCount.get() + 1);
+
+                                if(isFirstLine.get()) {
+                                    isFirstLine.set(false);
+                                }
+                            }
+
+                        } catch (final SuperCsvCellProcessorException e) {
```

Comment:
This should probably also catch "SuperCsvException". I ran into one when the number of columns I had as my processor list didn't match the number passed.


![screen shot 2016-07-27 at 7 35 46 pm](https://cloud.githubusercontent.com/assets/11302527/17196264/5594b3c2-5431-11e6-8014-b8f0743aa107.png)